sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -309,13 +314,25 @@ def test_imputation_copy():
     Xt.data[0] = -1
     assert_false(np.all(X.data == Xt.data))
 
+    X = X_orig_knn.copy().toarray()
+    imputer = KNNImputer(missing_values=0, copy=True)
```

Comment:
In merging the tests, I intended that you might just use a loop or something...

Reply:
creating a loop creates a conflict. Can this even be fixed at my end?

<<<<<<< knnimpute
=======
    # copy=False, dense => no copy
    X = X_orig.copy().toarray()
    imputer = Imputer(missing_values=0, strategy="mean", copy=False)
    Xt = imputer.fit(X).transform(X)
    Xt[0, 0] = -1
    assert_array_almost_equal(X, Xt)

>>>>>>> master

Reply:
Yes it can. Simply fix it up to be how it should look (all that was changed here was to use assert_array_almost_equal).

Ping @lesteve: this is why fixing problems when they're not broken may not be worthwhile.

Reply:
if you meant "use assert_array_almost_equal" instead, that did not work. i don't think you mean "unloop the test". so i guess i am at a loss here...